framework/Optimizers/Optimizer.py

```diff
@@ -620,229 +309,158 @@ def checkConvergence(self):
-        paramDict[key] = self.inputInfo[key]
-      else:
-        for var in self.inputInfo['SampledVars'].keys():
-          paramDict['Variable: '+var+' has value'] = paramDict[key][var]
-    paramDict.update(self.localGetCurrentSetting())
-    return paramDict
-
-  def getInitParams(self):
+      collects the objective variable from a realization and adjusts the sign for min/max
+      @ In, rlz, dict, realization particularly including objective variable
+      @ Out, optVal, float, sign-adjust objective value
     """
-      This function is called from the base class to print some of the information inside the class.
-      Whatever is permanent in the class and not inherited from the parent class should be mentioned here
-      The information is passed back in the dictionary. No information about values that change during the simulation are allowed
-      @ In, None
-      @ Out, paramDict, dict, dictionary containing the parameter names as keys
-                              and each parameter's initial value as the dictionary values
-    """
-    paramDict = {}
-    for variable in self.getOptVars():
-      paramDict[variable] = 'is sampled as a decision variable'
-    paramDict['limit_mdlEval' ]        = self.limit['mdlEval']
-    paramDict['limit_optIter']         = self.limit['varsUpdate']
-    paramDict['initial seed' ]         = self.initSeed
-    paramDict.update(self.localGetInitParams())
-    return paramDict
+    optVal = (-1 if self._minMax == 'max' else 1) * rlz[self._objectiveVar]
+    return optVal
 
-  @abc.abstractmethod
-  def _getJobsByID(self):
+  def _initializeInitSampler(self, externalSeeding):
     """
-      Overwritten by the base class; obtains new solution export values
-      @ In, None
+      Initializes samplers to be used for seeding trajectories.
+      @ In, externalSeeding, int, unused
       @ Out, None
     """
-    pass
-
-  def getLossFunctionGivenId(self, evaluationID):
-    """
-      Method to get the Loss Function value given an evaluation ID
-      @ In, evaluationID, string, the evaluation identifier (prefix)
-      @ Out, objeciveValue, float, the loss function value
-    """
-    # get matching realization by matching "prefix"
-    # TODO the EnsembleModel prefix breaks this pattern!
-    _,rlz  = self.mdlEvalHist.realization(matchDict={'prefix':evaluationID})
-    # if no match found, return None
-    if rlz is None:
-      return None
-    # otherwise, return value (float assures single value)
-    return float(rlz[self.objVar])
-
-  def getOptVars(self):
-    """
-      Returns the variables in the active optimization space
-      @ In, None
-      @ Out, optVars, list(string), variables in the current optimization space
-    """
-    return self.fullOptVars
-
-  def _incrementCounter(self):
-    """
-      Increments counter and sets up prefix.
-      @ In, None
+    if not self.assemblerDict.get('Sampler', False):
+      return
+    sampler = self.assemblerDict['Sampler'][0][3]
```

Comment:
personally, I think we should avoid to use the dict, but use method  retrieveObjectFromAssemblerDict instead. @alfoa

Reply:
I followed the existing implementation examples; we should correct this in a widespread sense if that is the intended behavior.

Reply:
I think when we touch the code, we'd better to change it if we think there is a better to do it.

Reply:
That may be true as long as it doesn't delay the release of critical infrastructure. A good fix-week option would be to standardize interactions with the Assembler throughout the code. If required, I can make this change here once I know what the proper assembler API methods to use are.

Reply:
This involves a lot more rework the make work across inheriting samplers than it's worth right now, in my opinion.